### DISCUSSION TITLE
Why `exists` is guarded by `i-implement-a-third-party-backend-and-opt-into-breaking-changes` ?

### DISCUSSION THREAD

**Discussion by vnghia** [diesel-rs/diesel#3936] [Q&A] [answered]
As title, I am just curious and is there any plan to move it to `diesel` public api ?

**Comment by weiznich** [comment]
I'm not 100% sure to which `exists` item you refer, therefore I will try to address all of them:

* `dsl::exists` (the function + the type def) should be part of the public API. If it's not shown as that it's a bug and we would appreciate a PR fixing that.
* `expressions::dsl::exists` (the function + the type def), same as above
* `expressions::exists` (the module and the content). These are not part of the public API as they contain implementation details that might change at a later point.

**Reply by vnghia** [reply]
I'm refering specifically to this one: https://docs.diesel.rs/master/diesel/dsl/fn.exists.html

**Reply by weiznich** [reply, marked as answer]
As written above: If that's shown as guarded by `i-implement-a-third-party-backend-and-opt-into-breaking-changes` that's a bug and we would like to accept a PR fixing that.

**Reply by vnghia** [reply]
I will send a PR soon
